File: run_pipeline.py
from __future__ import annotations

import argparse
import json
from pathlib import Path

from src.pipeline import run_pipeline


def parse_args() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description="Multi-source Candidate Data Transformer")
    parser.add_argument("--recruiter-csv", type=str, default=None)
    parser.add_argument("--ats-json", type=str, default=None)
    parser.add_argument("--notes", type=str, default=None)
    parser.add_argument("--config", type=str, default=None, help="Optional custom projection config JSON")
    parser.add_argument("--out", type=str, default="outputs/default_output.json")
    parser.add_argument("--candidate-id", type=str, default="candidate-001")
    return parser.parse_args()


def main() -> None:
    args = parse_args()
    custom_config = None
    if args.config:
        custom_config = json.loads(Path(args.config).read_text(encoding="utf-8"))

    result = run_pipeline(
        recruiter_csv_path=args.recruiter_csv,
        ats_json_path=args.ats_json,
        notes_txt_path=args.notes,
        config=custom_config,
        candidate_id=args.candidate_id,
    )

    out_path = Path(args.out)
    out_path.parent.mkdir(parents=True, exist_ok=True)
    out_path.write_text(json.dumps(result["projected"], indent=2), encoding="utf-8")

    print("Projected profile:")
    print(json.dumps(result["projected"], indent=2))
    if result["validation_errors"]:
        print("\nValidation errors:")
        for err in result["validation_errors"]:
            print("-", err)
    else:
        print("\nValidation: OK")


if __name__ == "__main__":
    main()

File: src/run_pipeline.ts
import { program } from 'commander';
import fs from 'fs';
import path from 'path';
import { runPipeline } from './pipeline';

const packageJson = JSON.parse(fs.readFileSync(path.join(__dirname, '../package.json'), 'utf-8'));

program
  .version(packageJson.version)
  .description('Multi-source Candidate Data Transformer')
  .option('--recruiter-csv <path>', 'Path to recruiter CSV export')
  .option('--ats-json <path>', 'Path to ATS JSON blob')
  .option('--notes <path>', 'Path to recruiter notes TXT')
  .option('--config <path>', 'Optional custom projection config JSON')
  .option('--out <path>', 'Output file path', 'outputs/default_output.json')
  .option('--candidate-id <id>', 'Candidate ID', 'candidate-001')
  .parse(process.argv);

const opts = program.opts();

try {
  let customConfig = null;
  if (opts.config) {
    const configContent = fs.readFileSync(opts.config, 'utf-8');
    customConfig = JSON.parse(configContent);
  }

  const result = runPipeline(opts.recruiterCsv, opts.atsJson, opts.notes, customConfig, opts.candidateId);

  const outDir = path.dirname(opts.out);
  if (!fs.existsSync(outDir)) {
    fs.mkdirSync(outDir, { recursive: true });
  }

  fs.writeFileSync(opts.out, JSON.stringify(result.projected, null, 2), 'utf-8');

  console.log('Projected profile:');
  console.log(JSON.stringify(result.projected, null, 2));

  if (result.validation_errors.length > 0) {
    console.log('\nValidation errors:');
    for (const err of result.validation_errors) {
      console.log('-', err);
    }
  } else {
    console.log('\nValidation: OK');
  }
} catch (err) {
  console.error('Error:', err);
  process.exit(1);
}
